File: Array/Easy/second_largest_element.java
public class second_largest_element {
    public static void main(String[] args) {
        int[] arr = {3, 5, 7, 2, 8};
        int secondLargest = findSecondLargest(arr);
        System.out.println("The second largest element in the array is: " + secondLargest);
    }

    public static int findSecondLargest(int[] arr) {
        // tc: O(n) - we need to traverse the entire array once to find the second largest element
        // sc: O(1) - we are using a constant amount of extra space to store the largest and second largest elements
        int largest = Integer.MIN_VALUE;
        int secondLargest = Integer.MIN_VALUE;

        for (int i = 0; i < arr.length; i++) {
            if (arr[i] > largest) {
                secondLargest = largest;
                largest = arr[i];
            } else if (arr[i] > secondLargest && arr[i] != largest) {
                secondLargest = arr[i];
            }
        }
        return secondLargest;
    }
}

File: Array/Easy/second_largest_element.cpp
#include <iostream>
using namespace std;
int getSecondLargest(vector<int> &arr)
{
    int largest = INT_MIN;
    int second_largest = INT_MIN;
    for (int i = 0; i < arr.size(); i++)
    {
        if (arr[i] > largest)
        {
            second_largest = largest;
            largest = arr[i];
        }
        else if (arr[i] > second_largest && arr[i] < largest)
        {
            second_largest = arr[i];
        }
    }
    return second_largest == INT_MIN ? -1 : second_largest;
}